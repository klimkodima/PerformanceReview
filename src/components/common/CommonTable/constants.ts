import { t } from 'i18next';

import { Order } from './types';

export const ascendingSort: Order = t('common_table.ascending_sort');
export const descendingSort: Order = t('common_table.descending_sort');
export const teamLeadBonus = t('common_table.team_lead_bonus');

export const HEAD_CELL_NAMES = {
  nameColumn: t('common_table.name_column'),
  pointsColumn: t('common_table.points_column'),
  performanceColumn: t('common_table.performance_column'),
  validityColumn: t('common_table.validity_column'),
  supportColumn: t('common_table.support_column'),
  correspondenceColumn: t('common_table.correspondence_column')
};

export enum CELL_COLOR_STYLE {
  minValueStyle = 'comparison-table-cell min-value-color',
  maxValueStyle = 'comparison-table-cell max-value-color',
  commonValueStyle = 'comparison-table-cell'
}

export const VALUE_TYPES = {
  task: t('common_table.task'),
  tasks: t('common_table.tasks'),
  percent: t('common_table.percent'),
  dash: t('common_table.dash'),
  hours: t('common_table.hours')
};
