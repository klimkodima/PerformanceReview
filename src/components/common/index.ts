export { ContentImage } from './ContentAuditorCommon';
export { ErrorTooltip } from './ErrorTooltip';
export { InfoTooltip } from './InfoTooltip';
export { WebsiteGroupsAccordion } from './WebsiteGroupsAccordion';
export { CommonTable } from '../common/CommonTable';
export { Modal } from './Modal';
export { Search } from './Search';
export { ConfirmationModal } from './ConfirmationModal';
export { NotificationText } from './ConfirmationModal';
