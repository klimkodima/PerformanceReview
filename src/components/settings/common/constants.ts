export const TABLE_BUTTONS: TableBtnsType = {
  editBtn: 'Edit Table',
  addBtn: 'Add',
  deleteBtn: 'Delete',
  saveBtn: 'Save',
  cancelBtn: 'Cancel'
};

type TableBtnsType = {
  [key: string]: string;
};
