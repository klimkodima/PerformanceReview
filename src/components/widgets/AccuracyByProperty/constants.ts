import { t } from 'i18next';

import { TableHeaderCellsType } from './types';

export const HEADER_CELLS: TableHeaderCellsType[] = [
  { name: t('accuracy_by_property.property_label'), isShowIcon: false },
  { name: t('accuracy_by_property.score_label'), isShowIcon: true }
];
