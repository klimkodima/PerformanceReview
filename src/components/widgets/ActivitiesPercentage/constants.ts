import { t } from 'i18next';

export const ACTIVITIES_PERCENTAGE_TITLE: string = t(
  'activities_percentage.widget_title_text'
);
