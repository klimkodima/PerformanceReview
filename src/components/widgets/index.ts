export { ActivitiesPercentage } from './ActivitiesPercentage';
export { ContentAuditorWidget } from './ContentAuditor';
export { PerformanceStatisticContainer } from './PerformanceStatistics';
export { TeamActivitiesPercentage } from './TeamActivitiesPercentage';
export { PerformanceStatisticsTeamLead } from './PerformanceStatisticsHOC';
export { PerformanceStatisticsAuditors } from './PerformanceStatisticsHOC';
export { CriteriaWidgets } from './CriteriaWidgets';
export { PointsWidget } from './TotalPointsWidget';
export { AuditorsStatistics } from './AuditorsStatistics';
export { TeamComparisonManager, TeamComparisonLead } from './TeamComparison';
export { AverageAccuracy } from './AverageAccuracy';
export { AccuracyByProperty } from './AccuracyByProperty';
