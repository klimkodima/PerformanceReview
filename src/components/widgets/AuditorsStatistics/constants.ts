import { t } from 'i18next';

import { HeadCellType } from '../../common/CommonTable/types';

export const headerName = t('auditors_statistics.header_name');

export const headCells: HeadCellType[] = [
  {
    id: t('common_table.name_column'),
    label: t('auditors_statistics.auditor_label')
  },
  {
    id: t('common_table.points_column'),
    label: t('auditors_statistics.points_label')
  },
  {
    id: t('common_table.performance_column'),
    label: t('auditors_statistics.performance_label')
  },
  {
    id: t('common_table.validity_column'),
    label: t('auditors_statistics.validity_label')
  },
  {
    id: t('common_table.support_column'),
    label: t('auditors_statistics.support_label')
  },
  {
    id: t('common_table.correspondence_column'),
    label: t('auditors_statistics.correspondence_label')
  }
];
