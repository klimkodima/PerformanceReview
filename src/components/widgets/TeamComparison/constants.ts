import { t } from 'i18next';

import { HeadCellType } from '../../common/CommonTable/types';

export const headerName = t('team_comparison.header_name');

export const headCells: HeadCellType[] = [
  {
    id: t('common_table.name_column'),
    label: t('team_comparison.team_label')
  },
  {
    id: t('common_table.points_column'),
    label: t('team_comparison.points_label')
  },
  {
    id: t('common_table.performance_column'),
    label: t('team_comparison.performance_label')
  },
  {
    id: t('common_table.validity_column'),
    label: t('team_comparison.validity_label')
  },
  {
    id: t('common_table.correspondence_column'),
    label: t('team_comparison.correspondence_label')
  }
];

export const PLACEHOLDER_TEXT = t('team_comparison.placeholder_text');
