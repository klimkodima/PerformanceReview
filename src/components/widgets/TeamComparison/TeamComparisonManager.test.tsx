import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { I18nextProvider } from 'react-i18next';

import store from 'src/store';
import i18n from 'src/i18n';
import { CommonTable } from 'src/components/common';
import { headCells, headerName } from './constants';

const mockTeamsData = [
  {
    leadName: 'Alphaname',
    name: 'Alpha',
    points: 25,
    performance: 3,
    validity: 100,
    correspondence: 97
  },
  {
    leadName: 'Charliename',
    name: 'Charlie',
    points: 13,
    performance: 1,
    validity: 100,
    correspondence: 97
  }
];

describe('TeamComparisonManager test', () => {
  test('TeamComparisonManager empty widget', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <I18nextProvider i18n={i18n}>
            <CommonTable
              data={[]}
              isTeamComparisonLead={false}
              checkedTeams={[]}
              defaultOrderBy={'name'}
              headCells={headCells}
              placeholderText={'Widget is empty'}
              headerName={headerName}
            />
          </I18nextProvider>
        </Provider>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('TeamComparisonManager 2 rows table', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <I18nextProvider i18n={i18n}>
            <CommonTable
              data={mockTeamsData}
              isTeamComparisonLead={false}
              checkedTeams={['Charlie', 'Alpha']}
              defaultOrderBy={'name'}
              headCells={headCells}
              placeholderText={''}
              headerName={headerName}
            />
          </I18nextProvider>
        </Provider>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
