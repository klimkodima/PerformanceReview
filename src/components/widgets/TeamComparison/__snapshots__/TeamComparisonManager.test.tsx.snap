// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamComparisonManager test TeamComparisonManager 2 rows table 1`] = `
<div
  className="comparison-table-container"
  data-testid="comparison-table-container"
>
  <div
    className="comparison-table-wrapper"
  >
    <h2
      className="comparison-table-header"
      data-testid="comparison-table-header"
    >
      Team Comparison
    </h2>
    <div
      className="MuiTableContainer-root comparison-table css-rorn0c-MuiTableContainer-root"
    >
      <table
        className="MuiTable-root MuiTable-stickyHeader comparison-table-content css-xn82ks-MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root comparison-table-head css-15wwp11-MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-head-cell css-dwuj3p-MuiTableCell-root"
              data-testid="comparison-table-head-cell"
              scope="col"
            >
              <span
                className="MuiButtonBase-root MuiTableSortLabel-root Mui-active css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                TEAM
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                  data-testid="ArrowDownwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-head-cell css-dwuj3p-MuiTableCell-root"
              data-testid="comparison-table-head-cell"
              scope="col"
            >
              <span
                className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                AVERAGE POINTS
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                  data-testid="ArrowDownwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-head-cell css-dwuj3p-MuiTableCell-root"
              data-testid="comparison-table-head-cell"
              scope="col"
            >
              <span
                className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                AVERAGE PERFORMANCE
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                  data-testid="ArrowDownwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-head-cell css-dwuj3p-MuiTableCell-root"
              data-testid="comparison-table-head-cell"
              scope="col"
            >
              <span
                className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                AVERAGE VALIDITY
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                  data-testid="ArrowDownwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-head-cell css-dwuj3p-MuiTableCell-root"
              data-testid="comparison-table-head-cell"
              scope="col"
            >
              <span
                className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                AVERAGE HOURS CORRESPONDENCE
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                  data-testid="ArrowDownwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-hover comparison-table-row css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-cell css-1yhpg23-MuiTableCell-root"
              data-testid="comparison-table-name-cell"
            >
              Alpha
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-cell max-value-color css-1yhpg23-MuiTableCell-root"
              data-testid="comparison-table-points-cell"
            >
              25
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-cell max-value-color css-1yhpg23-MuiTableCell-root"
              data-testid="comparison-table-performance-cell"
            >
              3
               tasks/h
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-cell max-value-color css-1yhpg23-MuiTableCell-root"
              data-testid="comparison-table-validity-cell"
            >
              100
              %
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-cell max-value-color css-1yhpg23-MuiTableCell-root"
              data-testid="comparison-table-correspondence-cell"
            >
              97
              %
            </td>
          </tr>
          <tr
            className="MuiTableRow-root MuiTableRow-hover comparison-table-row css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-cell css-1yhpg23-MuiTableCell-root"
              data-testid="comparison-table-name-cell"
            >
              Charlie
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-cell min-value-color css-1yhpg23-MuiTableCell-root"
              data-testid="comparison-table-points-cell"
            >
              13
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-cell min-value-color css-1yhpg23-MuiTableCell-root"
              data-testid="comparison-table-performance-cell"
            >
              1
               task/h
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-cell max-value-color css-1yhpg23-MuiTableCell-root"
              data-testid="comparison-table-validity-cell"
            >
              100
              %
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium comparison-table-cell max-value-color css-1yhpg23-MuiTableCell-root"
              data-testid="comparison-table-correspondence-cell"
            >
              97
              %
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`TeamComparisonManager test TeamComparisonManager empty widget 1`] = `
<div
  className="comparison-table-container"
  data-testid="comparison-table-container"
>
  <div
    className="comparison-table-wrapper"
  >
    <h2
      className="comparison-table-header"
      data-testid="comparison-table-header"
    >
      Team Comparison
    </h2>
    <div
      className="comparison-table-placeholder"
    >
      <p>
        Widget is empty
      </p>
    </div>
  </div>
</div>
`;
